roles/realm_join/tasks/realm_join_ad.yml

```
@@ -0,0 +1,121 @@
+---
+- name: Ensure pexpect to answer password
+  become: yes
+  package:
+    name: pexpect
+    state: latest
+  # on Fedora this will be True due to the major version. Not nice, but works.
+  when: ansible_os_family != "RedHat" or ansible_distribution_major_version >= '8'
```

```suggestion
  when: ansible_os_family != "RedHat" or ansible_distribution_major_version is version('8', '>')
```

But there is no `pexpect` on my Fedora 33, only `python3-pexpect`. And neither is there such a package on Debian ;-)